# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocket.h
# Commit: da41f0fab427
# Full Hash: da41f0fab427b199f1a0c6156467a0ad9a9e6891
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Make proxy config parameters optional in WebrtcTCPSocket to get the NAT simulator working again. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45104
# ==============================================================================

diff -r b83124cf3a89 -r da41f0fab427 media/mtransport/ipc/WebrtcTCPSocket.h
--- a/media/mtransport/ipc/WebrtcTCPSocket.h	Tue Sep 17 18:17:24 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocket.h	Tue Sep 17 18:17:37 2019 +0000
@@ -54,9 +54,9 @@
   void SetTabId(dom::TabId aTabId);
   nsresult Open(const nsCString& aHost, const int& aPort,
                 const nsCString& aLocalAddress, const int& aLocalPort,
-                bool aUseTls, const net::LoadInfoArgs& aArgs,
-                const nsCString& aAlpn,
-                NrSocketProxyConfig::ProxyPolicy aProxyPolicy);
+                bool aUseTls, const Maybe<net::LoadInfoArgs>& aArgs,
+                const Maybe<nsCString>& aAlpn,
+                const Maybe<NrSocketProxyConfig::ProxyPolicy>& aProxyPolicy);
   nsresult Write(nsTArray<uint8_t>&& aBytes);
   nsresult Close();
 